swift/ql/lib/codeql/swift/elements/expr/OtherConstructorDeclRefExprConstructor.qll

```diff
@@ -1,4 +1,8 @@
-// generated by codegen/codegen.py, remove this comment if you wish to edit this file
 private import codeql.swift.generated.Raw
+private import codeql.swift.elements.expr.MethodLookupExprConstructor
 
-predicate constructOtherConstructorDeclRefExpr(Raw::OtherConstructorDeclRefExpr id) { any() }
+predicate constructOtherConstructorDeclRefExpr(Raw::OtherConstructorDeclRefExpr id) {
+  // exclude an argument that will be part of a SelfApplyExpr
+  // that will be transformed into a MethodLookupExpr
+  not exists(Raw::SelfApplyExpr e | id.getConstructorDecl() = extractDeclFromSelfApplyExpr(e))
```

Comment:
Same goes here as I wrote for `DeclRefExprConstructor.qll`, we don't really want to hide any kind of `DeclRefExpr`, and leave it as is under `MethodLookupExpr::getMethod`.